Android/app/src/main/java/com/example/android/activities/BasketActivity.java

```diff
@@ -0,0 +1,215 @@
+        attachSwipeToDelete(recyclerView);
+    }
+
+    private void createCartAdapter(RecyclerView recyclerView) {
+        cartAdapter = new CartAdapter(this, CartManager.getInstance().getCartItems(), this);
+        recyclerView.setAdapter(cartAdapter);
+    }
+
+    private void attachSwipeToDelete(RecyclerView recyclerView) {
+        ItemTouchHelper itemTouchHelper = new ItemTouchHelper(new SwipeToDeleteCallback());
+        itemTouchHelper.attachToRecyclerView(recyclerView);
+    }
+
+    @SuppressLint("DefaultLocale")
+    private void updateTotalAmount() {
+        textViewTotal.setText(String.format("Total: $%.2f", cartAdapter.calculateTotal()));
+    }
+
+    @Override
+    public void onCartUpdated(double total) {
+        updateTotalAmount();
+        emptyBasketMessage.setVisibility(cartAdapter.getItemCount() == 1 ? View.VISIBLE : View.GONE);
+
+    }
+
+    private class SwipeToDeleteCallback extends ItemTouchHelper.SimpleCallback {
+        private static final int SWIPE_DIRECTION = ItemTouchHelper.RIGHT;
+        private static final String SWIPE_BACKGROUND_COLOR = "#FF8686";
+        private static final float SWIPE_THRESHOLD = 0.3f;
+
+        SwipeToDeleteCallback() {
+            super(0, SWIPE_DIRECTION);
+        }
+
+        @Override
+        public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
+            return false;
+        }
+
+        @Override
+        public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
+            int position = viewHolder.getBindingAdapterPosition();
+            if (position > 0) {
+                cartAdapter.removeItem(position);
+            } else {
+                cartAdapter.notifyItemChanged(0);
+            }
+        }
+
+        @Override
+        public int getSwipeDirs(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder) {
+            return viewHolder.getBindingAdapterPosition() == 0 ? 0 : super.getSwipeDirs(recyclerView, viewHolder);
+        }
+
+        @Override
+        public void onChildDraw(@NonNull Canvas c, @NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, float dX, float dY, int actionState, boolean isCurrentlyActive) {
+            if (isSwiping(actionState, viewHolder)) {
+                if (handleSwipeAnimation(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive))
+                    return;
+            }
+            super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive);
+        }
+
+        private boolean handleSwipeAnimation(@NonNull Canvas c, @NonNull RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, float dX, float dY, int actionState, boolean isCurrentlyActive) {
+            float swipeThreshold = calculateSwipeThreshold(viewHolder);
+
+            if (isSwipingRight(dX)) {
```

Comment:
maybe you could somehow "hide this statements" like extract them to new method, 'cause they look bad here :D